File: src/main/java/com/example/librarymanagement/service/PublisherService.java
package com.example.librarymanagement.service;

import com.example.librarymanagement.model.entity.Publisher;

import java.util.List;

public interface PublisherService {
    Publisher createPublisher(Publisher publisher);
    Publisher updatePublisher(Long id, Publisher updatedPublisher);
    Publisher getPublisherById(Long id);
    List<Publisher> getAllPublishers();
}

File: src/main/java/com/example/librarymanagement/service/PublisherService.kt
package com.example.librarymanagement.service

import com.example.librarymanagement.model.domain.Publisher

interface PublisherService {
    fun createPublisher(publisher: Publisher): Publisher
    fun updatePublisher(updatedPublisher: Publisher): Publisher
    fun getPublisherById(id: String): Publisher
    fun getAllPublishers(): List<Publisher>
}
